src/formatter.rs

```diff
@@ -10,44 +10,113 @@ lazy_static! {
     static ref RE_CONSTANT: Regex = Regex::new(r"\{\{((?s).*?)\}\}").unwrap();
     static ref RE_BOLD: Regex = Regex::new(r"<<((?s).*?)>>").unwrap();
     static ref RE_MONOSPACE: Regex = Regex::new(r"`([^`]*?)`").unwrap();
-    static ref RE_MONOSPACE_TRIM: Regex = Regex::new(r"\n? *(`[^`]*`) *").unwrap();
+    static ref RE_MONOSPACE_OLD: Regex = Regex::new(r"```([^`]*?)```").unwrap();
+    static ref RE_MONOSPACE_TRIM: Regex = Regex::new(r"\s*`\s*([^`]+?)\s*`\s*").unwrap();
     static ref RE_BACKTICK: Regex = Regex::new(r"(`[^`]+`)|([^`]+)").unwrap();
     static ref RE_SPACES: Regex = Regex::new(r" +").unwrap();
     static ref RE_NONWHITESPACE: Regex = Regex::new(r"[^\r\n ]+").unwrap();
+    static ref RE_NEWLINES: Regex = Regex::new(r"\n\n\n+").unwrap();
+    static ref RE_ALL_BUT_MONOSPACE: Regex =
+        Regex::new(r"\[\[((?s).*?)\]\]|\{\{((?s).*?)\}\}|<<((?s).*?)>>").unwrap();
 }
 
-/// Format Codingame statement that contains special formatting syntax
-/// [[VARIABLE]] - {{CONSTANT}} - <<BOLD>> - `MONOSPACE`
 pub fn format_cg(text: &str, ostyle: &OutputStyle) -> String {
-    // Trim consecutive spaces (imitates html behaviour)
-    // But only if it's not in a Monospace block (between backticks ``)
-    let mut result = RE_BACKTICK
-        .replace_all(text, |caps: &regex::Captures| {
-            if let Some(backtick_text) = caps.get(1) {
-                backtick_text.as_str().to_string()
-            } else if let Some(non_backtick_text) = caps.get(2) {
-                RE_SPACES.replace_all(non_backtick_text.as_str(), " ").to_string()
+    let mut text = format_edit_monospace(&text, ostyle);
+    text = format_trim_consecutive_spaces(&text);
+    text = format_monospace_padding(&text);
+    text = format_add_reverse_nester_tags(&text);
+    text = format_paint_inner_blocks(&text, ostyle);
+    format_remove_excessive_newlines(&text)
+}
+
+/// 1. Throws a warning if it finds outdated formatting
+/// 2. Replaces ```text``` -> `text`
+/// 3. Format whitespace around Monospace blocks.
+///
+/// Clashes with outdated formatting:
+///     https://www.codingame.com/contribute/view/25623694f80d8f747b3fa474a33a9920335ce
+///     https://www.codingame.com/contribute/view/7018d709bf39dcccec4ed9f97fb18105f64c
+/// Others:
+///     https://www.codingame.com/contribute/view/1222536cec20519e1a630ecc8ada367dd708b
```

Comment:
In my experience the chance of them being changed or removed is extremely low. 

But even if they were, the links are there to see how they display in the terminal, rather than testing particular bits of logic, so I'm unsure on how to proceed. Should we have a folder with clashes for testing? This may relate to the clash database idea of #37 

What do you suggest to do?

Reply:
There's no need to have the full context of the clash, a test case with the relevant part of the statement is enough.

Reply:
But how does the relevant part of the statement get tested for _how good they look in the terminal_? This is not something I can test with only the relevant part of the statement. I would need to run `clash show` on some clash that either is already in the database, or somewhere else just for testing purposes.